wwwapp/views.py

```diff
@@ -195,10 +195,10 @@ def workshop_view(request, name=None):
         else:
             has_perm_to_edit = True
     else:
-        workshop = Workshop.objects.get(name=name)
+        workshop = get_object_or_404(Workshop, name=name)
         title = workshop.title
         if request.user.is_authenticated:
```

Comment:
You don't need to check if user is authenticated even, this is checked by `can_edit_workshop`